src/SingleReport/BootMouse.cpp

```diff
@@ -73,18 +78,24 @@ int BootMouse_::getInterface(uint8_t* interfaceCount)
 
 int BootMouse_::getDescriptor(USBSetup& setup)
 {
-	// Check if this is a HID Class Descriptor request
-	if (setup.bmRequestType != REQUEST_DEVICETOHOST_STANDARD_INTERFACE) { return 0; }
-	if (setup.wValueH != HID_REPORT_DESCRIPTOR_TYPE) { return 0; }
-
 	// In a HID Class Descriptor wIndex cointains the interface number
 	if (setup.wIndex != pluggedInterface) { return 0; }
 
-	// Reset the protocol on reenumeration. Normally the host should not assume the state of the protocol
-	// due to the USB specs, but Windows and Linux just assumes its in report mode.
-	protocol = HID_REPORT_PROTOCOL;
+	// Check if this is a HID Class Descriptor request
+	if (setup.bmRequestType != REQUEST_DEVICETOHOST_STANDARD_INTERFACE) { return 0; }
+
+	if (setup.wValueH == HID_HID_DESCRIPTOR_TYPE) {
+		// Apple UEFI wants it
+		HIDDescDescriptor desc = D_HIDREPORT(sizeof(_hidReportDescriptorMouse));
+		return USB_SendControl(0, &desc, sizeof(desc));
```

Comment:
Do we need this for the bootkeyboard as well?

Reply:
Totally yes. This is the second problem that I described in the issue. This is how hardware mice behave, I also found it in the datasheet for some [mouse chip](https://www.silabs.com/documents/public/application-notes/AN249.pdf). Almost all the devices I have personally met do not require this command, but all Macs do. It's funny that this constant exists in the HID library, but is not used. Apparently, no one thought that the command associated with it could be mandatory.

Reply:
So I guess you will be trying to fix the keyboard in another PR? I think this PR is ready to merge then, I dont see anything critical that should not work.

Reply:
Oh, crap, I'm sorry, I missed that you asked about the keyboard, and continued about the mouse. Stupid me. No, apparently this is not necessary for the keyboard. At least Apple and everything else works. But, POTENTIALLY, this may be due to that bug with KVM and strange BIOSes. Maybe if you add this code, everything will work. In the analysis of USB data, I have not seen a reference to this command, and I have not yet been able to find a dataset for some keyboard chip where this would be described. I would ask someone to experiment, but it's definitely not worth including this for the keyboard until we're sure.